src/providers/elixir.rs

```diff
@@ -0,0 +1,61 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::Environment,
+    nix::pkg::Pkg,
+    plan::{
+        phase::{Phase, StartPhase},
+        BuildPlan,
+    },
+};
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct DenoTasks {
+    pub start: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct DenoJson {
+    pub tasks: Option<DenoTasks>,
+}
+
+pub struct ElixirProvider {}
+
+impl Provider for ElixirProvider {
+    fn name(&self) -> &str {
+        "elixir"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("mix.exs"))
+    }
+
+    fn get_build_plan(&self, _app: &App, _env: &Environment) -> Result<Option<BuildPlan>> {
+        let mut plan = BuildPlan::default();
```

Comment:
bit nit: The plan can be created with all the phases at the bottom of this function with

```rs
let plan = BuildPlan::new(vec![setup_phase, install_phase, build_phase], Some(start_phase));
```

It does the same thing but it a few lines shorter.

Reply:
Tried to do that but kept getting a borrow exception